packages/rpc-methods/src/restricted/getBip32Entropy.ts

```diff
@@ -0,0 +1,228 @@
+import {
+  constructPermission,
+  PermissionFactory,
+  PermissionSpecificationBuilder,
+  PermissionType,
+  RestrictedMethodOptions,
+  ValidPermissionSpecification,
+  CaveatSpecificationConstraint,
+  Caveat,
+} from '@metamask/controllers';
+import { ethErrors } from 'eth-rpc-errors';
+import { NonEmptyArray } from '@metamask/utils';
+import { BIP32Node, JsonSLIP10Node, SLIP10Node } from '@metamask/key-tree';
+
+import { SnapCaveatType } from '../caveats';
+
+const DERIVATION_PATH_REGEX = /^m(\/[0-9]+'?)+$/u;
+
+const targetKey = 'snap_getBip32Entropy';
+
+export type GetBip32EntropyMethodHooks = {
+  /**
+   * @returns The mnemonic of the user's primary keyring.
+   */
+  getMnemonic: () => Promise<string>;
+
+  /**
+   * Waits for the extension to be unlocked.
+   *
+   * @returns A promise that resolves once the extension is unlocked.
+   */
+  getUnlockPromise: (shouldShowUnlockRequest: boolean) => Promise<void>;
+};
+
+type GetBip32EntropySpecificationBuilderOptions = {
+  allowedCaveats?: Readonly<NonEmptyArray<string>> | null;
+  methodHooks: GetBip32EntropyMethodHooks;
+};
+
+type GetBip32EntropySpecification = ValidPermissionSpecification<{
+  permissionType: PermissionType.RestrictedMethod;
+  targetKey: typeof targetKey;
+  methodImplementation: ReturnType<typeof getBip32EntropyImplementation>;
+  allowedCaveats: Readonly<NonEmptyArray<string>> | null;
+}>;
+
+type GetBip32EntropyParameters = {
+  path: string;
+  curve?: 'secp256k1' | 'ed25519';
+};
+
+// TODO: See if this is necessary.
+const getBip32EntropyFactory: PermissionFactory<any, any> = (
+  permissionOptions,
+) => {
+  return constructPermission(permissionOptions);
+};
+
+/**
+ * Validate a path object.
+ *
+ * @param value - The value to validate.
+ * @throws If the value is invalid.
+ */
+function validatePath(
+  value: unknown,
+): asserts value is GetBip32EntropyParameters {
+  // TODO: Use `@metamask/utils`. The extension currently uses an old version of
+  // `utils`, so this is just a workaround for testing.
+  const isPlainObject = (v: unknown): v is Record<string, unknown> => {
+    return true;
+  };
+
+  if (!isPlainObject(value)) {
+    throw ethErrors.rpc.invalidParams({
+      message: 'Expected a plain object.',
+    });
+  }
```

Comment:
Same issue as above, current version of `utils` used in the extension doesn't have this function.